### ISSUE TITLE
Better categorization of FMU results

### ISSUE THREAD

**Issue by perolavsvendsen** [equinor/fmu-dataio#291]
This is combining several issues and discussions around both consumption and production of FMU results. Generally, data produced by FMU can be categorized in several different ways. Examples:

- Standardized (formal?) results: Data that are generally produced by all FMU workflows, where strict standards should apply. Producers should be expected to follow strict standards, and consumers should expect stability and predictability. Examples of such results could be (but not limited to): `inplace volumes`, `production profiles`, `structural horizons in depth`, `zone thickness in depth`, `fault intersection lines`, `field outline`, `fluid contact` and others - to be defined.
- Non-standard results: Modeling is highly customized, and the flexibility to produce any results at any point in time is crucial. Non-standard results can be "promoted" to standard results.

For standardized results, fmu-dataio could take a bigger role in enforcing the standard.

Further, data produced by FMU workflows can be separated into "predictions" and "QC" output. This creates significant challenges for consumption if methods for separating the two are not available. Example: A specific depth surface is exported 10 times through a workflow, but only one of those is the actual prediction - the others are exported for QC purposes.

In short:
- There is a need for clearly separating predictions and QC output.
- There is a need for defining standard outputs and enforcing strict standard.
- There is a need for supporting non-standard outputs.

Through metadata, we need to facilitate a better filters on what data within an FMU case is to be piped further to various consumers. experience so far is that consumers are overwhelmed when exposed to _all_ output. 

Example user stories:
- As an FMU consumer, I would like to present my users with The Prediction of a specific horizon in depth from FMU, so that my user can safely utilize FMU results out of context.
- As an FMU consumer, I would like to query The Inplace Volumes from an FMU case, so that I avoid confusion and erroneous information.

**Comment by perolavsvendsen** [comment]
#140 #183 #200 #220

**Comment by perolavsvendsen** [comment]
This is (more or less) the implementation of Standard Results, so perhaps this is now stale and can be closed. The requirements and the user stories are still valid:

- There is a need for clearly separating predictions and QC output.
- There is a need for defining standard outputs and enforcing strict standard.
- There is a need for supporting non-standard outputs.

As an FMU data consumer, I would like to present my users with The Prediction of a specific horizon in depth from FMU, so that my user can safely utilize FMU results out of context. _The point here is that for a consumer who does not know anything about a specific model, it will not be possible to understand which data to use if predictions and QC output are mixed together._

As an FMU data consumer, I would like to query The Inplace Volumes from an FMU case, so that I avoid confusion and erroneous information. _Same point as above - we might export lots of (different) inplace volumes from a model, but external consumers won't know which one is The Actual Inplace Volumes from this model if all are mixed together._
